### ISSUE TITLE
Prompt for missing required parameters

### ISSUE THREAD

**Issue by marcelbrueckner** [BrianPugh/cyclopts#143]
How to implement Typer's [CLI Option Prompt](https://typer.tiangolo.com/tutorial/options/prompt/) with cyclopts?
Is there a solution other than inspecting the tokens via [Meta App](https://cyclopts.readthedocs.io/en/latest/meta_app.html)?

In my specific use-case, I want to ask for a password if it's not passed as parameter (which usually shouldn't be passed in clear text via CLI, but I want to let the users decide).

**Comment by BrianPugh** [comment]
Cyclopts purposely limited its responsibility-scope to just the initial command CLI parsing. Other libraries are quite good at prompting the user for more data. My personal favorite is [Questionary](https://github.com/tmbo/questionary). With Questionary, your program may look like:

```
import questionary
from cyclopts import App

app = App()

@app.default
def main(password: Optional[str] = None):
    if password is None:
        password = questionary.password("What's your password?").ask()
    # Do stuff with the password here.
```

**Comment by marcelbrueckner** [comment]
Oh, that was fast :) Thanks. I thought about that already, but then the nice "required" hint in the CLI's help isn't displayed anymore.

However, I've now experimented with the Parameters and discovered that I can set `required=True`.
This way (in combination with setting a default value) I have the corresponding hint in the CLI's help,
but without the error when the parameter isn't passed :)

So your code finally resembles to the following:

```python
from cyclopts import App
# I will use rich instead of questionary because cyclopts depends on rich anyway
from rich.prompt import Prompt

app = App()

@app.default
def main(password: Annotated[Optional[str], Parameter(required=True)] = None):
    if password is None:
        password = Prompt.ask("What's your password?", password=True)
    # Do stuff with the password here.
```

**Comment by BrianPugh** [comment]
Thanks for the use-case example!

> However, I've now experimented with the Parameters and discovered that I can set required=True.
This way (in combination with setting a default value) I have the corresponding hint in the CLI's help,
but without the error when the parameter isn't passed :)

I think this behavior is technically a bug in Cyclopts's side, so I can't exactly recommend on relying on that behavior. I'll have to investigate more later when I have a moment. I'm going to keep this open until I provide some more feedback.

**Comment by marcelbrueckner** [comment]
> I personally don't think it's a valid use-case to mark an item as required in the help-page if it's not actually required at invocation time. However I would definitely like to hear your opinion!

Given the user wants to invoke my (future) tool in a non-interactive environment, the parameter needs to be passed at invocation time. Hence for me, that's "required" and I would like to express that in my CLI help in a visually coherent way.

For example, other (sub)commands of my tool will require non-password'ish parameters that I don't ask for interactively if missing. They will have the red "required" hint in CLI's help and I want the same for the password parameter.

**Comment by BrianPugh** [comment]
I updated the docs and fixed a minor related bug in #146. This is now part of the v2.5.1 release. Thank you!
